### PR TITLE
Dse fixes

### PR DESCRIPTION
**Pull request by ptnapoleon**
@kishkaru and @mambocab to review, please. CI is running on this now, as far as C* goes.

### CODE REVIEW

File: ccmlib/cluster.py [hunk 1]
```diff
@@ -318,10 +318,6 @@ def start(self, no_wait=False, verbose=False, wait_for_binary_proto=False, wait_
                     if old_node is not node:
                         old_node.watch_log_for_alive(node, from_mark=mark)
 
-        if wait_for_binary_proto:
```
Comment (ptnapoleon) [inline ccmlib/cluster.py]: This worries me, and I think I'd like to undo this change, assuming DSE still works w/o it.

File: ccmlib/dse_node.py [hunk 2]
```diff
@@ -66,8 +66,8 @@ def start(self,
               wait_other_notice=False,
               replace_token=None,
               replace_address=None,
-              jvm_args=None,
-              wait_for_binary_proto=False,
+              jvm_args=[],
```
Comment (ptnapoleon) [inline ccmlib/dse_node.py]: This jvm_args change is clearly wrong and needs fixed.

File: ccmlib/node.py [hunk 3]
```diff
@@ -775,13 +776,18 @@ def run_cqlsh(self, cmds=None, show_output=False, cqlsh_options=None, return_out
             for cmd in cmds.split(';'):
                 cmd = cmd.strip()
                 if cmd:
-                    cmd_str += (cmd + ';\n')
-            cmd_str += "quit;\n"
-            output = p.communicate(input=cmd_str)
+                    p.stdin.write(cmd + ';\n')
+            p.stdin.write("quit;\n")
+            p.wait()
```
Comment (ptnapoleon) [inline ccmlib/node.py]: This line change doesn't seem right, as now we no longer have `output` defined.

**Comment by ptnapoleon** [comment]
Okay this is ready for review from the two of you. I made some changes, as this broke some dtests.

**Comment by kishkaru** [comment]
Works great, tested on latest 4.7 and 4.8.

**Comment by ptnapoleon** [comment]
Cassci just got back to me. Looks like this is failing some dtests.

File: ccmlib/dse_node.py [hunk 4]
```diff
@@ -67,7 +67,7 @@ def start(self,
               replace_token=None,
               replace_address=None,
               jvm_args=None,
-              wait_for_binary_proto=False,
+              wait_for_binary_proto=True,
```
Comment (mambocab) [inline ccmlib/dse_node.py]: Worried about this API change; is it necessary? I don't know anything about pcmanus/ccm's DSE userbase, fwiw.

File: ccmlib/dse_node.py [hunk 5]
```diff
@@ -95,6 +95,8 @@ def start(self,
         if wait_other_notice:
             marks = [(node, node.mark_log()) for node in list(self.cluster.nodes.values()) if node.is_running()]
 
+        self.mark = self.mark_log()
```
Comment (mambocab) [inline ccmlib/dse_node.py]: Any particular reason to make this `self.mark` and not just `mark`? I worry about random state on a started node being an attractive nuisance.

File: ccmlib/dse_node.py [hunk 6]
```diff
@@ -180,7 +178,10 @@ def start(self,
                 node.watch_log_for_alive(self, from_mark=mark)
 
         if wait_for_binary_proto:
-            self.wait_for_binary_interface()
+            self.watch_log_for("Starting listening for CQL clients", from_mark=self.mark)
```
Comment (mambocab) [inline ccmlib/dse_node.py]: Are we moving away from `wait_for_binary_interface` so we can use the mark? I'd rather add a `mark` parameter to `wait_for_binary_interface` than duplicate the `"Starting listening for CQL clients"`. Alternatively we could define that string as a shared constant somewhere.

**Comment by mambocab** [comment]
Seems reasonable outside my comments.

File: ccmlib/dse_node.py (hunk 6 above)
Comment (ptnapoleon) [inline ccmlib/dse_node.py]: It already has a mark parameter, afaik. I believe it may have been to stop checking if the socket is listening.

File: ccmlib/dse_node.py (hunk 5 above)
Comment (ptnapoleon) [inline ccmlib/dse_node.py]: It's self.mark in node.py currently. We can hash out the fix on the main branch there, but symmetry is preferred for now.

File: ccmlib/node.py [hunk 7]
```diff
@@ -378,7 +378,7 @@ def watch_log_for(self, exprs, from_mark=None, timeout=600, process=None, verbos
                     if process.returncode != 0:
                         raise RuntimeError()  # Shouldn't reuse RuntimeError but I'm lazy
 
-        with open(log_file) as f:
+        with open(self.logfilename()) as f:
```
Comment (ptnapoleon) [inline ccmlib/node.py]: This change is what broke all of the tests.

**Comment by ptnapoleon** [comment]
Okay, works for launching DSE, and I have responded to all of Jim's comments. Assuming CI is happy with the latest run, I will merge this.
